dts/msm8974/msm8974pro-htc-m8.dts

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+/dts-v1/;
+
+#include <skeleton.dtsi>
+#include <lk2nd.h>
+
+/ {
+	/* Actually htc,project-id */
+	qcom,msm-id = <271 0 0x10000>, <271 1 0x10000>, <271 2 0x10000>,
+		      <271 3 0x10000>, <271 4 0x10000>, <271 5 0x10000>,
+		      <272 0 0x10000>, <272 1 0x10000>, <272 2 0x10000>,
+		      <272 3 0x10000>, <272 4 0x10000>, <272 5 0x10000>,
+		      <273 0 0x10000>, <273 1 0x10000>, <273 2 0x10000>,
+		      <273 3 0x10000>, <273 4 0x10000>, <273 5 0x10000>,
+		      <280 0 0x10000>, <280 1 0x10000>, <280 2 0x10000>,
+		      <280 3 0x10000>, <280 4 0x10000>, <280 5 0x10000>,
+		      <286 1 0x10000>, <286 2 0x10000>, <286 3 0x10000>,
+		      <286 4 0x10000>, <286 5 0x10000>,
+		      <305 0 0x10000>, <305 1 0x10000>, <305 2 0x10000>,
+		      <305 3 0x10000>, <305 4 0x10000>, <305 5 0x10000>;
+
+	htc-m8 {
+		model = "HTC One M8";
+		compatible = "htc,m8", "qcom,msm8974", "lk2nd,device";
+		lk2nd,match-panel;
+
+		lk2nd,keys =
+			<KEY_VOLUMEDOWN	27 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>,
+			<KEY_VOLUMEUP	28 (GPIO_PULL_UP | GPIO_ACTIVE_LOW)>;
+
+		panel {
+			compatible = "htc,m8-panel";
+
+			qcom,mdss_dsi_glu_sharp_1080p_cmd {};
```

Comment:
IMO it would be nice to assign compatibles to those already, even if we don't know if any other are actually used.

Reply:
I added some compatibles, but I don't know if they are correct and if they will be the same that will be used in the Linux kernel. Let me know if I should change them!

Reply:
FWIW It's an arbitrary choice and later when/if we need to use panel selection, we can use those compatibles when creating/generating panel drivers. I.e. for msm8916 we generate drivers for known tested panels with a special tool, and assign compatibles in it's config like here:

https://github.com/msm8916-mainline/linux-panel-drivers/blob/921a9179af5624a6980d2f9572b8ebe4ce3e8fc7/config/wingtech-wt88047.sh#L1-L12